content/doc/developer/architecture/extensions.adoc

```diff
@@ -2,4 +2,10 @@
 title: Extensions
 layout: developersection
 wip: true
----
\ No newline at end of file
+---
+
+Jenkins defines extension points, which are interfaces or abstract classes that model an aspect of its behavior. Those interfaces define contracts of what need to be implemented, and Jenkins allows plugins to contribute those implementations. In general, all you need to do to register an implementation is to mark it with `@Extension`.
```

Comment:
```suggestion
Jenkins defines extension points, which are interfaces or abstract classes that model an aspect of its behavior.
The interface or class defines contracts of what needs implementation, and Jenkins allows plugins to contribute those implementations.
To register an implementation, make sure you mark it with `@Extension`.
```
A small suggestion to update the content to be one sentence per line (making it easier to read and review), and adjust some of the wording to flow a bit better.